tests/test_scalars.py

```diff
@@ -0,0 +1,663 @@
+    ge = check(assert_type(c2 >= td, bool), bool)
+    assert lt != ge
+
+    lt_b = check(assert_type(c3 < td, Any), np.bool_)
+    ge_b = check(assert_type(c3 >= td, Any), np.bool_)
+    assert lt_b != ge_b
+
+    lt_a = check(assert_type(c4 < td, np_ndarray_bool), np.ndarray)
+    ge_a = check(assert_type(c4 >= td, np_ndarray_bool), np.ndarray)
+    assert (lt_a != ge_a).all()
+
+    lt_a = check(assert_type(c5 < td, np_ndarray_bool), np.ndarray)
+    ge_a = check(assert_type(c5 >= td, np_ndarray_bool), np.ndarray)
+    assert (lt_a != ge_a).all()
+
+    eq_s = check(assert_type(c6 < td, "pd.Series[bool]"), pd.Series, bool)
+    ne_s = check(assert_type(c6 >= td, "pd.Series[bool]"), pd.Series, bool)
+    assert (eq_s != ne_s).all()
+
+    eq = check(assert_type(td == td, bool), bool)
+    ne = check(assert_type(td != td, bool), bool)
+    assert eq != ne
+
+    eq = check(assert_type(td == c2, bool), bool)
+    ne = check(assert_type(td != c2, bool), bool)
+    assert eq != ne
+
+    eq = check(assert_type(td == c3, bool), bool)
+    ne = check(assert_type(td != c3, bool), bool)
+    assert eq != ne
+
+    eq_a = check(assert_type(td == c4, np_ndarray_bool), np.ndarray)
+    ne_a = check(assert_type(td != c4, np_ndarray_bool), np.ndarray)
+    assert (eq_a != ne_a).all()
+
+    eq_a = check(assert_type(td == c5, np_ndarray_bool), np.ndarray)
+    ne_a = check(assert_type(td != c5, np_ndarray_bool), np.ndarray)
+    assert (eq_a != ne_a).all()
+
+    eq_s = check(assert_type(td == c6, "pd.Series[bool]"), pd.Series, bool)
+    ne_s = check(assert_type(td != c6, "pd.Series[bool]"), pd.Series, bool)
+    assert (eq_s != ne_s).all()
+
+    eq = check(assert_type(c2 == td, bool), bool)
+    ne = check(assert_type(c2 != td, bool), bool)
+    assert eq != ne
+
+    eq = check(assert_type(c3 == td, Any), np.bool_)
+    ne = check(assert_type(c3 != td, Any), np.bool_)
+    assert eq != ne
+
+    eq_a = check(assert_type(c4 == td, Any), np.ndarray)
+    ne_a = check(assert_type(c4 != td, Any), np.ndarray)
+    assert (eq_a != ne_a).all()
+
+    eq_a = check(assert_type(c5 == td, np_ndarray_bool), np.ndarray)
+    ne_a = check(assert_type(c5 != td, np_ndarray_bool), np.ndarray)
+    assert (eq_a != ne_a).all()
+
+    eq_s = check(assert_type(c6 == td, "pd.Series[bool]"), pd.Series, bool)
+    ne_s = check(assert_type(c6 != td, "pd.Series[bool]"), pd.Series, bool)
+    assert (eq_s != ne_s).all()
+
+    eq = check(assert_type(td == 1, bool), bool)
+    ne = check(assert_type(td != 1, bool), bool)
+    assert eq != ne
+
+    eq = check(assert_type(td == (3 + 2j), bool), bool)
+    ne = check(assert_type(td != (3 + 2j), bool), bool)
+    assert eq != ne
+
+    eq_s = check(
+        assert_type(td == pd.Series([1, 2, 3]), "pd.Series[bool]"), pd.Series, bool
+    )
+    ne_s = check(
+        assert_type(td != pd.Series([1, 2, 3]), "pd.Series[bool]"), pd.Series, bool
+    )
+    assert (eq_s != ne_s).all()
+
+
+def test_timestamp_types_init() -> None:
```

Comment:
shouldn't this be in the `Timestamp` PR?

Same for the various test funcs below here.

Reply:
Moved these